src/Sparkfabrik/Command/Gitlab/GitlabMergeRequestCommand.php

```diff
@@ -159,11 +159,19 @@ private function handleArguments(InputInterface $input, OutputInterface $output,
      *
      * @param string|integer $project_id
      *
-     * @return integer|boolean
+     * @return integer| ConsoleOutput output.
      */
     private function handleAgumentProjectId($project_id = null)
     {
-        return ($project_id ? $project_id : $this->getService()->getConfig()['project_id']);
+        $conf_project_id = $this->getService()->getConfig()['project_id'];
+        if (isset($project_id) && !is_integer($project_id)) {
+            $project_id = $this->findProjectId($project_id);
+        } else if (isset($conf_project_id) &&
+        !is_integer($conf_project_id)) {
+            $project_id = $this->findProjectId($conf_project_id);
+        } else {
+            return ($project_id ? $project_id : $conf_project_id);
+        }
```

Comment:
because 'test' isn't a string, will be called the 'findProjectId' function that find all projects with the 'test' string in their names and output a list to get the ID. The output will be prompted also if we have only a result to avoid a double api call for each command